c/datasetjson.c

```diff
@@ -1756,7 +1759,27 @@ void respondWithDatasetMetadata(HttpResponse *response) {
 
   int fieldCount = defaultCSIFieldCount;
   char **csiFields = defaultCSIFields;
-
+  int asteriskPos = indexOf(dsn,dsnLen,'*',0);
```

Comment:
You might want to check if addQualifiers is on before you start getting positions in the string. You may be doing work for no reason. You can do all of these calculations inside the if statement. I don't see you using them anywhere else in the code.